Getting errors while trying out schemathesis.

I installed `schemathesis` using `pip3 install schemathesis` and tried it out as follows:

```bash
$ st run http://localhost:8000/graphql  --show-errors-tracebacks
/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/errors.py:127: HypothesisDeprecationWarning: MultipleFailures is deprecated; use the builtin `BaseExceptionGroup` type instead, or `exceptiongroup.BaseExceptionGroup` before Python 3.11
  note_deprecation(
============================================================================= Schemathesis test session starts =============================================================================
Schema location: http://localhost:8000/graphql
Base URL: http://localhost:8000/graphql
Specification version: GraphQL
Workers: 1
Collected API operations: 4

MyQuery.resolverOne /home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/errors.py:127: HypothesisDeprecationWarning: MultipleFailures is deprecated; use the builtin `BaseExceptionGroup` type instead, or `exceptiongroup.BaseExceptionGroup` before Python 3.11
  note_deprecation(
E                                                                                                                                                        [ 25%]
MyQuery.resolverTwo /home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/errors.py:127: HypothesisDeprecationWarning: MultipleFailures is deprecated; use the builtin `BaseExceptionGroup` type instead, or `exceptiongroup.BaseExceptionGroup` before Python 3.11
  note_deprecation(
E                                                                                                                                               [ 50%]
MyQuery.resolverThree /home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/errors.py:127: HypothesisDeprecationWarning: MultipleFailures is deprecated; use the builtin `BaseExceptionGroup` type instead, or `exceptiongroup.BaseExceptionGroup` before Python 3.11
  note_deprecation(
E                                                                                                                                                     [ 75%]
MyQuery.resolverFour /home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/errors.py:127: HypothesisDeprecationWarning: MultipleFailures is deprecated; use the builtin `BaseExceptionGroup` type instead, or `exceptiongroup.BaseExceptionGroup` before Python 3.11
  note_deprecation(
E                                                                                                                                                          [100%]

========================================================================================== ERRORS ==========================================================================================
_____________________________________________________________________________ MyQuery.resolverOne [P] _____________________________________________________________________________
Traceback (most recent call last):
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 307, in run_test
    test(checks, targets, result, errors=errors, headers=headers, **kwargs)
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 570, in network_test
    case: Case,
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/core.py", line 1257, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/specs/graphql/schemas.py", line 208, in get_case_strategy
    strategy = {
KeyError: 'MyQuery'

________________________________________________________________________ MyQuery.resolverTwo [P] _________________________________________________________________________
Traceback (most recent call last):
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 307, in run_test
    test(checks, targets, result, errors=errors, headers=headers, **kwargs)
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 570, in network_test
    case: Case,
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/core.py", line 1257, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/specs/graphql/schemas.py", line 208, in get_case_strategy
    strategy = {
KeyError: 'MyQuery'

___________________________________________________________________________ MyQuery.resolverThree [P] ____________________________________________________________________________
Traceback (most recent call last):
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 307, in run_test
    test(checks, targets, result, errors=errors, headers=headers, **kwargs)
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 570, in network_test
    case: Case,
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/core.py", line 1257, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/specs/graphql/schemas.py", line 208, in get_case_strategy
    strategy = {
KeyError: 'MyQuery'

______________________________________________________________________________ MyQuery.resolverFour [P] ______________________________________________________________________________
Traceback (most recent call last):
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 307, in run_test
    test(checks, targets, result, errors=errors, headers=headers, **kwargs)
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/runner/impl/core.py", line 570, in network_test
    case: Case,
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/hypothesis/core.py", line 1257, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/hbarthwal/mypython-venv/lib/python3.8/site-packages/schemathesis/specs/graphql/schemas.py", line 208, in get_case_strategy
    strategy = {
KeyError: 'MyQuery'


If you need assistance in solving this, feel free to join our Discord server and report it:

  https://discord.gg/R9ASRAmHnA

========================================================================================= SUMMARY ==========================================================================================

No checks were performed.

Hint: You can visualize test results in Schemathesis.io by using `--report` in your CLI command.

==================================================================================== 4 errored in 0.07s ====================================================================================
```
Can someone tell me how to fix these errors?

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -7,6 +7,7 @@ Changelog
 **Fixed**
 
 - CLI: Hanging on ``CTRL-C`` when ``--report`` is enabled.
+- Internal error when GraphQL schema has its root types renamed. `#1591`_
 
 .. _v3.16.4:
 
@@ -3045,6 +3046,7 @@ Deprecated
 .. _0.3.0: https://github.com/schemathesis/schemathesis/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/schemathesis/schemathesis/compare/v0.1.0...v0.2.0
 
+.. _#1591: https://github.com/schemathesis/schemathesis/issues/1591
 .. _#1584: https://github.com/schemathesis/schemathesis/issues/1584
 .. _#1582: https://github.com/schemathesis/schemathesis/issues/1582
 .. _#1580: https://github.com/schemathesis/schemathesis/issues/1580

--- a/src/schemathesis/specs/graphql/schemas.py
+++ b/src/schemathesis/specs/graphql/schemas.py
@@ -1,5 +1,7 @@
 # Pylint bug - `Callable` is used below
 # pylint: disable=unused-import
+import enum
+from enum import unique
 from typing import Any, Callable, Dict, Generator, List, Optional, Sequence, Tuple, Type, TypeVar, Union, cast
 from urllib.parse import urlsplit
 
@@ -23,6 +25,12 @@ from ...stateful import Stateful, StatefulTest
 from ...types import Body, Cookies, Headers, NotSet, PathParameters, Query
 from ...utils import NOT_SET, GenericResponse, Ok, Result
 from .scalars import CUSTOM_SCALARS
+
+
+@unique
+class RootType(enum.Enum):
+    QUERY = enum.auto()
+    MUTATION = enum.auto()
 
 
 @attr.s(slots=True, repr=False)  # pragma: no mutate
@@ -82,6 +90,7 @@ C = TypeVar("C", bound=Case)
 class GraphQLOperationDefinition(OperationDefinition):
     field_name: str = attr.ib()
     type_: graphql.GraphQLType = attr.ib()
+    root_type: RootType = attr.ib()
 
 
 @attr.s()  # pragma: no mutate
@@ -124,7 +133,10 @@ class GraphQLSchema(BaseSchema):
 
     def get_all_operations(self) -> Generator[Result[APIOperation, InvalidSchema], None, None]:
         schema = self.client_schema
-        for operation_type in (schema.query_type, schema.mutation_type):
+        for root_type, operation_type in (
+            (RootType.QUERY, schema.query_type),
+            (RootType.MUTATION, schema.mutation_type),
+        ):
             if operation_type is None:
                 continue
             for field_name, definition in operation_type.fields.items():
@@ -144,6 +156,7 @@ class GraphQLSchema(BaseSchema):
                             parameters=[],
                             type_=operation_type,
                             field_name=field_name,
+                            root_type=root_type,
                         ),
                         case_cls=GraphQLCase,
                     )
@@ -206,9 +219,9 @@ def get_case_strategy(
 ) -> Any:
     definition = cast(GraphQLOperationDefinition, operation.definition)
     strategy = {
-        "Query": gql_st.queries,
-        "Mutation": gql_st.mutations,
-    }[definition.type_.name]
+        RootType.QUERY: gql_st.queries,
+        RootType.MUTATION: gql_st.mutations,
+    }[definition.root_type]
     body = draw(strategy(client_schema, fields=[definition.field_name], custom_scalars=CUSTOM_SCALARS))
     instance = GraphQLCase(body=body, operation=operation, data_generation_method=data_generation_method)  # type: ignore
     context = auth.AuthContext(

--- a/test/specs/graphql/test_basic.py
+++ b/test/specs/graphql/test_basic.py
@@ -83,3 +83,30 @@ def test_operations_count(graphql_url):
     raw_schema = decoded["data"]
     schema = schemathesis.graphql.from_dict(raw_schema)
     assert schema.operations_count == 4
+
+
+def test_type_names():
+    # When the user gives custom names to query types
+    raw_schema = """
+    schema {
+       query: MyQuery
+       mutation: MyMutation
+    }
+
+    type MyQuery {
+       v: String
+    }
+    type MyMutation {
+       v(i: Int): String
+    }
+    """
+    # Then the schema should be loaded without errors
+    schema = schemathesis.graphql.from_file(raw_schema)
+    # And requests should be properly generated
+
+    @given(case=schema[b""]["POST"].as_strategy())
+    @settings(max_examples=1, deadline=None)
+    def test(case):
+        pass
+
+    test()

